aiida/cmdline/commands/cmd_import.py

```diff
@@ -36,13 +37,108 @@ class ExtrasImportCode(Enum):
+            extras_mode_new=kwargs['extras_mode_new'],
+            comment_mode=kwargs['comment_mode'])
+    except exceptions.IncompatibleArchiveVersionError as exception:
+        if first:
+            # Migration has not yet been tried.
+            if kwargs['migration']:
+                # Confirm migration
+                echo.echo_warning(str(exception).splitlines()[0])
+                if kwargs['non_interactive']:
+                    migrate_archive = True
+                else:
+                    migrate_archive = click.confirm(
+                        "Do you want to try and migrate {} to the newest export file version?\n"
+                        "Note: This will not change your current file.".format(kwargs['archive']),
+                        default=True,
+                        abort=True)
+            else:
+                # Abort
+                echo.echo_critical(str(exception))
+        else:
+            # Migration has been performed, something is still wrong
+            crit_message = "{} has been migrated, but it still cannot be imported.\n{}".format(
+                kwargs['archive'], exception)
+            echo.echo_critical(crit_message)
+    except Exception:
+        echo.echo_error('an exception occurred while importing the archive {}'.format(kwargs['archive']))
+        echo.echo(traceback.format_exc())
+        if not kwargs['non_interactive']:
+            click.confirm('do you want to continue?', abort=True)
+    else:
+        echo.echo_success('imported archive {}'.format(kwargs['archive']))
+
+    return migrate_archive
+
+
+def _migrate_archive(ctx, temp_folder, **kwargs):
+    """Utility functiuon for `verdi import` to migrate archive"""
+    from aiida.cmdline.commands.cmd_export import migrate
+
+    # Checks
+    expected_keys = ['file_to_import', 'archive', 'non_interactive']
+    for key in expected_keys:
+        if key not in kwargs:
+            raise ValueError("{} needed for utility function '{}'".format(key, '_migrate_archive'))
+
+    # Initialization
+    temp_out_file = 'migrated_importfile.aiida'
+
+    # Migration
+    try:
+        ctx.invoke(
+            migrate,
+            input_file=kwargs['file_to_import'],
+            output_file=temp_folder.get_abs_path(temp_out_file),
+            silent=False)
+    except Exception:
+        echo.echo_error("an exception occurred while migrating the archive {}.\n"
+                        "Use 'verdi export migrate' to update this export file.".format(kwargs['archive']))
+        echo.echo(traceback.format_exc())
+        if not kwargs['non_interactive']:
+            click.confirm('do you want to continue?', abort=True)
+    else:
+        echo.echo_success("archive migrated, proceeding with import")
+
+    kwargs['file_to_import'] = temp_folder.get_abs_path(temp_out_file)
+
+    # Try again to import archive
+    _try_import(first=False, **kwargs)
```

Comment:
to me, this does not belong in `_migrate_archive` but should be called explicitly.

Reply:
Okay, I agree now you bring it up. I will remove it out and put it in the click command.